src/main/java/com/theironyard/MicroblogSpringController.java

```diff
@@ -0,0 +1,48 @@
+package com.theironyard;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+
+@Controller
+public class MicroblogSpringController {
+    private static String loggedInUser = "loggedInUser";
+    ArrayList<Message> messages = new ArrayList<>();
+
+
+    @RequestMapping(path = "/add-message", method = RequestMethod.POST)
+    public String add(String text) {
+        Message msg = new Message(messages.size() + 1, text);
+        messages.add(msg);
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/delete-message", method = RequestMethod.POST)
+    public String delete(int id) {
+            messages.remove(id - 1);
```

Comment:
This will work sometimes. If you use the index and later remove messages with a lower index, then the id stored in the object will no longer align with the index -1.  So, consider how you can fix this to delete the exact message by it's id.